ethtx/providers/node/base.py

```diff
@@ -0,0 +1,27 @@
+#  Copyright 2021 DAI Foundation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at: http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+
+class NodeBase:
+    node: str
+    poa: bool
+
+    def __init__(self, node: str, poa: bool):
+        super().__init__()
+        self.node = node
+        self.poa = poa
+
+    def __repr__(self) -> str:
+        return f"<Node: {self.node}, poa: {self.poa}>"
+
+    def __iter__(self):
+        return iter(self.__dict__.items())
```

Comment:
1. This could be a dataclass, making it simpler
2. `node` variable name is super generic, is it node url? If yes please rename it to `url`. And would be useful to add `name` to it for debugging logs.
3. Is this `repr` and `iter` needed? Not sure about `repr`, but this is not a iterable object so `iter`  shouldn't be here